geode-core/src/test/java/org/apache/geode/internal/cache/versions/RegionVersionVectorJUnitTest.java

```diff
@@ -527,6 +538,61 @@ public void testRegionVersionInTags() {
     assertEquals("failed test for bug #48576", version, tag.getRegionVersion());
   }
 
+  @Test
+  public void testRecordVersionDuringRegionInit() {
+    LocalRegion mockRegion = mock(LocalRegion.class);
+    when(mockRegion.isInitialized()).thenReturn(false);
+    final String local = NetworkUtils.getIPLiteral();
+    InternalDistributedMember ownerId = new InternalDistributedMember(local, 101);
+    VMVersionTag tag = new VMVersionTag();
+    tag.setRegionVersion(1L);
+
+    RegionVersionVector rvv = createRegionVersionVector(ownerId, mockRegion);
+    rvv.recordVersion(ownerId, tag);
+    assertEquals(1, rvv.getVersionForMember(ownerId));
+  }
+
+  @Test
+  public void testRecordVersionAfterRegionInitThrowsException() {
+    expectedException.expect(InternalGemFireError.class);
```

Comment:
I generally like to place the expect statement just before the line that will cause the exception: that way you are being more explicit about where an exception can occur and the test cannot pass erroneously through an exception elsewhere than where you planned.

Reply:
Fixed in new Diff.